target/ppc/translate/vector-impl.c.inc

```diff
@@ -54,3 +54,19 @@ static bool trans_VCFUGED(DisasContext *ctx, arg_VX *a)
 
     return true;
 }
+
+static bool trans_VCFUGED(DisasContext *ctx, arg_VX *a)
```

Comment:
Should we REQUIRE_64BIT and wrap the call to do_call_vx_doubleword_helper with #if defined(TARGET_PPC64)/qemu_build_not_reached() too?

I've been meaning to ask you this before, but ended up forgetting to.

Without these 2, aren't we being less restrictive with vcfuged than with cfuged? (this applies to the other vector x non-vector instructions that share helpers too)

Reply:
The ifdef is not required here. It's used in cfuged/pdepd/pextd to avoid a compilation error for ppc-softmmu and ppc-linux-user targets, where `cpu_gpr` is `TCGv` (i.e. `TCGv_i32` for those targets). Even though the code after REQUIRE_64BIT is optimized, the compiler checks the types, and `gen_helper` receives `TCGv_i64`.

(We could avoid the error by declaring the helper receiving `target_ulong`, but IIRC vector registers are 128-bits even in 32-bits, so it would lead to wrong results.)

About REQUIRE_64BIT, I'm not sure. The PowerISA explicitly says that cfuged/pdepd/pextd is 64-bits only (in the section name), but I can't see the same for vcfuged/vpdepd/vpextd. It's reasonable that no 32-bits CPU implement them, but I don't know about 32-bit code on 64-bit CPU.

Reply:
To add to this discussion, I'm also not sure about REQUIRE_64BIT. Stuff like plfd, for example, isn't obvious that needs to be running in 64 bit mode to be correct. Could it be enough to add a requirement to the ISA 3.1?